apps/docs-e2e/src/docs.spec.ts

```diff
@@ -150,23 +154,287 @@ test.describe('should contain docs page elements', {
+  await page.setViewportSize({ width: 320, height: 480 })
+  await page.goto(DOCS_SITE_URL)
+  await expect(page).toHaveTitle(/Welcome to the Docs/)
+})
+
 test.describe('should contain mobile header elements', {
   tag: '@smoke',
 }, () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto(DOCS_SITE_URL)
+    await page.setViewportSize({ width: 320, height: 480 })
+  })
+
   test('should contain cuHacking Logo Icon in mobile header', async ({ docsLayoutPage }) => {
-    await docsLayoutPage.page.setViewportSize({
-      width: 320,
-      height: 480,
-    })
     await expect(docsLayoutPage.cuHackingLogoIcon).toBeVisible()
   })
 
   test('should contain hamburger icon in mobile header', async ({ docsLayoutPage }) => {
-    await docsLayoutPage.page.setViewportSize({
-      width: 320,
-      height: 480,
-    })
     await expect(docsLayoutPage.hamburgerIcon).toBeVisible()
   })
+
+  test('should take user to docs home page when cuHacking logo icon is clicked', async ({ docsLayoutPage }) => {
+    await docsLayoutPage.cuHackingLogoIcon.click()
+    await expect(docsLayoutPage.page).toHaveURL(DOCS_SITE_URL)
+  })
+
+  test('should contain cuHacking logo Text in mobile header', async ({ docsLayoutPage }) => {
+    await expect(docsLayoutPage.cuHackingLogoText).toBeVisible()
+  })
+
+  test('should contain search icon in mobile header', async ({ page }) => {
+    const searchIcon = page.getByRole('button', { name: 'Open Search' })
+    await expect(searchIcon).toBeVisible()
+  })
+
+  test('should show search modal when search icon in mobile header is clicked', async ({ page }) => {
+    await page.getByRole('button', { name: 'Open Search' }).click()
+    const searchModal = page.getByPlaceholder('Search')
+    await expect(searchModal).toBeVisible()
+  })
+
+  test('should have quicklinks mobile header', async ({ page }) => {
+    await expect(page.getByRole('button', { name: 'On this page Quick Links' })).toBeVisible()
+  })
+
+  test('should have edit on github button and link work', async ({ page }) => {
+    await page.getByRole('button', { name: 'On this page Quick Links' }).click()
+    const pagePromise = page.waitForEvent('popup')
+    await page.getByRole('link', { name: 'Edit on GitHub' }).click()
+    const newPage = await pagePromise
+    await expect(newPage).toHaveURL(DOCS_INDEX_PAGE_GITHUB_URL)
+  })
+})
+
+test.describe('should contain hamburger icon elements', {
+  tag: '@smoke',
+}, () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto(DOCS_SITE_URL)
+    await page.setViewportSize({ width: 320, height: 480 })
+    await page.getByLabel('Toggle Sidebar').click()
+  })
+
+  test('should contain sections dropdown in mobile hamburger icon', async ({ page }) => {
+    await expect(page.getByRole('button', { name: 'Home Leave none behind' })).toBeVisible()
+  })
+
+  test('should have website link and button work', async ({ page }) => {
```

Comment:
This is a visibility test mixed with an interaction test. We want to keep them separate. 

You can see how I did "should contain...", "should take user to .... when x action is done...", "should do y....". Hence **button work** is not very descriptive.